// angular2-simple-notify/notifyComponent.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require('@angular/core');
var router_1 = require('@angular/router');
var notifyService_1 = require("./notifyService");
var NotifyComponent = (function () {
    function NotifyComponent(_router) {
        this._router = _router;
        this.timeout = 3000;
        this.position = "top right";
        this.type = "success";
    }
    NotifyComponent.prototype.ngOnInit = function () {
        var _this = this;
        this._router.changes.subscribe(function () { return _this._getNotify(); });
        notifyService_1.NotifyService.timeout = this.timeout;
        this._getNotify();
    };
    NotifyComponent.prototype._getNotify = function () {
        notifyService_1.NotifyService.clearNotify();
        this.notify = notifyService_1.NotifyService.getAllNotify();
    };
    NotifyComponent.prototype._getStyle = function () {
        return {
            top: (/^top|top$/.test(this.position)) ? '0' : '',
            bottom: (/^bottom|bottom$/.test(this.position)) ? '0' : '',
            right: (/^right|right$/.test(this.position)) ? '0' : '',
            left: (/^left|left$/.test(this.position)) ? '0' : '',
            position: 'fixed',
            width: '400px',
            'max-width': '100vw',
            'z-index': '5100',
        };
    };
    ;
    __decorate([
        core_1.Input(), 
        __metadata('design:type', Number)
    ], NotifyComponent.prototype, "timeout", void 0);
    __decorate([
        core_1.Input(), 
        __metadata('design:type', String)
    ], NotifyComponent.prototype, "position", void 0);
    __decorate([
        core_1.Input(), 
        __metadata('design:type', String)
    ], NotifyComponent.prototype, "type", void 0);
    NotifyComponent = __decorate([
        core_1.Component({
            selector: 'notify',
            template: "\n<div [ngStyle]=\"_getStyle()\">\n    <div\n        *ngFor=\"let n of notify\"\n        [hidden]=\"!n.active\"\n        [innerHtml]=\"n.content\"\n        class=\"alert alert-dismissible alert-{{n.type || type}}\">\n    </div>\n</div>\n"
        }), 
        __metadata('design:paramtypes', [router_1.Router])
    ], NotifyComponent);
    return NotifyComponent;
}());
exports.NotifyComponent = NotifyComponent;
